src/main/options.js

```diff
@@ -29,30 +30,28 @@ function normalize(options, opts) {
     Object.assign({}, hiddenDefaults)
   );
 
-  if (opts.inferParser !== false) {
-    if (
-      rawOptions.filepath &&
-      (!rawOptions.parser || rawOptions.parser === defaults.parser)
-    ) {
-      const inferredParser = inferParser(
-        rawOptions.filepath,
-        rawOptions.plugins
+  if (!rawOptions.parser) {
+    const logger = (opts && opts.logger) || console;
+    if (!rawOptions.filepath) {
+      logger.warn("Skipping file with unknown filepath");
```

Comment:
What is an unknown filepath?

Reply:
@lydell Use stdin without `--stdin-filepath` or `prettier.format` without `{ filepath }`

Reply:
At first I decided to throw an error but I figured that could be a little too harsh

Reply:
I see. I wanted to come up with a better message, but it's hard.

Reply:
Really like the updated message, good job!